src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -36173,7 +36104,14 @@ public void UnconstrainedTypeParameter_Return_03()
     static U F20<T, U>(T t) where U : T, new() => (U)t;
 }";
             var comp = CreateCompilation(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition }, parseOptions: TestOptions.Regular8);
-            comp.VerifyDiagnostics();
+            comp.VerifyDiagnostics(
+                // (20,51): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //     static U F18<T, U>(T t) where U : class, T => (U)t;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(U)t").WithLocation(20, 51),
+                // prototype(NullableReferenceTypes): Should this be a warning, as there was already a nested warning about converting t to U?
```

Comment:
>prototype [](start = 19, length = 9)

`PROTOTYPE`

Reply:
Still active.

---
In reply to: [206389940](https://github.com/dotnet/roslyn/pull/28956#discussion_r206389940) [](ancestors = 206389940)